libvips/foreign/webpsave.c

```diff
@@ -746,6 +767,9 @@ vips_foreign_save_webp_build(VipsObject *object)
 		return -1;
 	}
 
+	if (!vips_object_argument_isset(object, "passes"))
+		webp->passes = vips_object_argument_isset(object, "target_size") ? TARGET_SIZE_DEFAULT_PASSES : 1;
```

Comment:
You don't need the `if()` if you're using a ternary.

Reply:
ie. I'd write:

```C
if (!vips_object_argument_isset(object, "passes"))
		webp->passes = 3;
```

(I think a hardwired number is OK here)

Reply:
I think there maybe is some misunderstanding here. `pass` _has_ an effect even if `target_size` isn't set. Changing it to be `3` here will cause an issue with making encoding slower for very little gain.

Increasing pass increases encoding time:

```
Starting passes=1
passes=1 completed, elapsed 0:00:01.128829
Size with passes=1: 656040

Starting passes=2
passes=2 completed, elapsed 0:00:01.599618
Size with passes=2: 655410

Starting passes=3
passes=3 completed, elapsed 0:00:02.295408
Size with passes=3: 655382

Starting passes=4
passes=4 completed, elapsed 0:00:03.015794
Size with passes=4: 655398

Starting passes=5
passes=5 completed, elapsed 0:00:03.729868
Size with passes=5: 655412

Starting passes=6
passes=6 completed, elapsed 0:00:04.411422
Size with passes=6: 655424

Starting passes=7
passes=7 completed, elapsed 0:00:05.138159
Size with passes=7: 655490

Starting passes=8
passes=8 completed, elapsed 0:00:05.855748
Size with passes=8: 655712

Starting passes=9
passes=9 completed, elapsed 0:00:06.545841
Size with passes=9: 655582

Starting passes=10
passes=10 completed, elapsed 0:00:07.225836
Size with passes=10: 655640
```

This is *without* a target_size set at all.

Exercise code:

```python3
with open("test.jpg", "rb") as f:
    buffer = f.read()

image = pyvips.Image.new_from_buffer(buffer, "")

for i in range(1, 11):
    with Bench(f"passes={i}"):
        buff = image.webpsave_buffer(passes=i)
    print(f"Size with passes={i}: {len(buff)}")
    print()
```

So if we set passes to three, it will increase encoding time for very minimal quality gains.